tests/integration/client/examples/example_keys_test.go

```diff
@@ -23,71 +23,89 @@ import (
 	"go.etcd.io/etcd/v3/client"
 )
 
+func mockKeysAPI_directory() {
+	// TODO: Replace with proper mocking
+	fmt.Println(`Key: "/myNodes/key1", Value: "value1"`)
+	fmt.Println(`Key: "/myNodes/key2", Value: "value2"`)
+}
+
 func ExampleKeysAPI_directory() {
-	c, err := client.New(client.Config{
-		Endpoints: exampleEndpoints,
-		Transport: exampleTransport,
-	})
-	if err != nil {
-		log.Fatal(err)
-	}
-	kapi := client.NewKeysAPI(c)
+	forUnitTestsRunInMockedContext(
+		mockKeysAPI_directory,
+		func() {
```

Comment:
Maybe define a new function `exampleKeysAPI_directory` for this inline function, next to `mockKeysAPI_directory`? I think it will be cleaner. What do you think?

Reply:
The doc-generator takes only 'body' of ExampleFoo function and puts it in the generated docs, see: 
  https://blog.golang.org/examples

So moving the content to separate function would make the examples not usable.